### DISCUSSION TITLE
how to authenticate against API to create a Role

### DISCUSSION THREAD

**Discussion by Masber** [keycloak/keycloak#31993] [Q&A]
Dear Keycloak community, 

I have a Keycloak instance and I am trying to learn how to create a "Role" using the API. 

I tried like this:

```
curl -H "Authorization: Bearer $TOKEN" 'https://api-gw-service.local/keycloak/admin/realms/myrealm/roles' \
-d '{"name":"test","description":"test","attributes":{}}'
```

Unfortunately, this is not works and I am getting an authorization error, however, I can login and create the role through the web interface using the same credentials.

My questions are: 

- how should I use curl to create a Role in keycloak?
- what is wrong in my curl command or with my token?

**Comment by danielFesenmeyer** [comment]
Probably your token is not correct.
You can find an example on how to create a token here: https://quarkus.io/guides/security-keycloak-authorization#testing

Also note, that the admin interface reads the roles from the database - for other clients, the roles are read from the token. Thus, when using a client different from security-admin-console or admin-cli, the roles must be in the scope specified when creating the token.

Furthermore, you should specify an additional Content-Type header: `-H "Content-Type: application/json" `
